vault/init.go

```diff
@@ -385,9 +375,11 @@ func (c *Core) Initialize(ctx context.Context, initParams *InitParams) (*InitRes
 		results.RootToken = base64.StdEncoding.EncodeToString(encryptedVals[0])
 	}
 
-	if err := c.createRaftTLSKeyring(ctx); err != nil {
-		c.logger.Error("failed to create raft TLS keyring", "error", err)
-		return nil, err
+	if raftBackend != nil {
+		if _, err := c.raftCreateTLSKeyring(ctx); err != nil {
```

Comment:
You've wrapped these functions in that nil check a few places, but don't they already check if we are running raft storage in the function?

Reply:
Yeah, it's not necessary to wrap in this case. The other call is based on `onInit` though, so I'll leave that untouched.

Reply:
Actually I've been trying to get away from returning `(nil, nil)` as a pattern. I'll keep this as-is and return (nil, error) instead on that method.